cabal-install/tests/UnitTests/Distribution/Solver/Modular/DSL.hs

```diff
@@ -246,59 +251,100 @@ type ExampleDb = [Either ExampleInstalled ExampleAvailable]
+                            C.setupDepends = mkSetupDeps deps,
+                            C.defaultSetupDepends = False
+                          }
+        package = SourcePackage {
+            packageInfoId        = pkgId
+          , packageSource        = LocalTarballPackage "<<path>>"
+          , packageDescrOverride = Nothing
+          , packageDescription   = C.GenericPackageDescription {
+                C.packageDescription = C.emptyPackageDescription {
+                    C.package        = pkgId
+                  , C.library        = error "not yet configured: library"
+                  , C.subLibraries   = error "not yet configured: subLibraries"
+                  , C.executables    = error "not yet configured: executables"
+                  , C.testSuites     = error "not yet configured: testSuites"
+                  , C.benchmarks     = error "not yet configured: benchmarks"
+                  , C.buildDepends   = error "not yet configured: buildDepends"
+                  , C.setupBuildInfo = setup
+                  , C.license = BSD3
+                  , C.buildType = if isNothing setup
+                                  then Just C.Simple
+                                  else Just C.Custom
+                  , C.category = "category"
+                  , C.maintainer = "maintainer"
+                  , C.description = "description"
+                  , C.synopsis = "synopsis"
+                  , C.licenseFiles = ["LICENSE"]
+                  , C.specVersionRaw = Left $ C.mkVersion [1,12]
+                  }
+              , C.genPackageFlags = nub $ concatMap extractFlags $
+                                    CD.libraryDeps (exAvDeps ex)
+                                     ++ concatMap snd testSuites
+                                     ++ concatMap snd executables
+              , C.condLibrary =
+                  let mkLib bi = mempty { C.libBuildInfo = bi }
+                  in Just $ mkCondTree defaultLib mkLib $ mkBuildInfoTree $
+                     Buildable (CD.libraryDeps (exAvDeps ex))
+              , C.condSubLibraries = []
+              , C.condExecutables =
+                  let mkTree = mkCondTree defaultExe mkExe . mkBuildInfoTree . Buildable
+                      mkExe bi = mempty { C.buildInfo = bi }
+                  in map (\(t, deps) -> (t, mkTree deps)) executables
+              , C.condTestSuites =
+                  let mkTree = mkCondTree defaultTest mkTest . mkBuildInfoTree . Buildable
+                      mkTest bi = mempty { C.testBuildInfo = bi }
+                  in map (\(t, deps) -> (t, mkTree deps)) testSuites
+              , C.condBenchmarks  = []
+              }
+            }
+        pkgCheckErrors =
+          let ignore = ["Unknown extensions:", "Unknown languages:"]
+          in [ err | err <- C.checkPackage (packageDescription package) Nothing
+             , not $ any (`isPrefixOf` C.explanation err) ignore ]
```

Comment:
I don't know.  If most extensions don't need special handling, and we can get their names directly from the compiler, then it seems like it would be simpler to not define all of them.

I'll just add a comment to the line above for now.